code/__HELPERS/game.dm

```diff
@@ -497,12 +497,16 @@
 
 //Recursively checks if an item is inside a given type, even through layers of storage. Returns the atom if it finds it.
 /proc/recursive_loc_check(atom/movable/target, type)
-	var/atom/A = target
-	while(!istype(A.loc, type))
-		if(!A.loc)
-			return
-		A = A.loc
-	return A
+  var/atom/A = target
+  if(istype(A, type))
+    return A
+
+  while(!istype(A.loc, type))
+    if(!A.loc)
+      return
+    A = A.loc
+
+  return A.loc

```

Comment:
Shouldn't this remain as `return A`?

Reply:
No, because the function is checking the type of A.loc, not A.  The function currently returns A, which is why it doesn't work.